src/pkg/services/m365/api/channels_pager.go

```diff
@@ -14,30 +21,205 @@ type ChannelMessageDeltaEnumerator interface {
 	SetNextLinker
 }
 
-// TODO: implement
-// var _ ChannelMessageDeltaEnumerator = &messagePageCtrl{}
+var _ ChannelMessageDeltaEnumerator = &MessagePageCtrl{}
+
+type MessagePageCtrl struct {
+	gs      graph.Servicer
+	builder *teams.ItemChannelsItemMessagesDeltaRequestBuilder
+	options *teams.ItemChannelsItemMessagesDeltaRequestBuilderGetRequestConfiguration
+}
+
+func (c Channels) NewMessagePager(
+	teamID,
+	channelID string,
+	fields []string,
+) *MessagePageCtrl {
+	requestConfig := &teams.ItemChannelsItemMessagesDeltaRequestBuilderGetRequestConfiguration{
+		QueryParameters: &teams.ItemChannelsItemMessagesDeltaRequestBuilderGetQueryParameters{
+			Select: fields,
+		},
+	}
+
+	res := &MessagePageCtrl{
+		gs:      c.Stable,
+		options: requestConfig,
+		builder: c.Stable.
+			Client().
+			Teams().
+			ByTeamId(teamID).
+			Channels().
+			ByChannelId(channelID).
+			Messages().
+			Delta(),
+	}
+
+	return res
+}
+
+func (p *MessagePageCtrl) SetNext(nextLink string) {
+	p.builder = teams.NewItemChannelsItemMessagesDeltaRequestBuilder(nextLink, p.gs.Adapter())
+}
+
+func (p *MessagePageCtrl) GetPage(ctx context.Context) (DeltaPageLinker, error) {
+	var (
+		resp DeltaPageLinker
+		err  error
+	)
+
+	resp, err = p.builder.Get(ctx, p.options)
+	if err != nil {
+		return nil, graph.Stack(ctx, err)
+	}
+
+	return resp, nil
+}
 
-// type messagePageCtrl struct {
-// 	gs      graph.Servicer
-// 	builder *teams.ItemChannelsItemMessagesRequestBuilder
-// 	options *teams.ItemChannelsItemMessagesRequestBuilderGetRequestConfiguration
-// }
+func (p *MessagePageCtrl) ValuesIn(l PageLinker) ([]models.ChatMessageable, error) {
+	return getValues[models.ChatMessageable](l)
+}
 
 // ---------------------------------------------------------------------------
 // channel pager
 // ---------------------------------------------------------------------------
 
 type ChannelDeltaEnumerator interface {
-	DeltaGetPager
+	PageLinker
 	ValuesInPageLinker[models.Channelable]
 	SetNextLinker
 }
 
 // TODO: implement
-// var _ ChannelDeltaEnumerator = &channelsPageCtrl{}
+var _ ChannelDeltaEnumerator = &channelPageCtrl{}
```

Comment:
Channels does not use a delta query query, and thus should not be named `delta` in any location.
```suggestion
var _ ChannelEnumerator = &channelPageCtrl{}
```